File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/SingleArgumentCommand.java
package com.github.bumblebee.command;

import telegram.domain.Update;
import telegram.polling.UpdateHandler;

public abstract class SingleArgumentCommand implements UpdateHandler {

    public abstract void handleCommand(Update update, Long chatId, String argument);

    @Override
    public boolean onUpdate(Update update) {
        Long chatId = update.getMessage().getChat().getId();
        String text = update.getMessage().getText();
        int cmdEndIndex = text.indexOf(' ');

        String argument = null;
        if (cmdEndIndex > 0 && cmdEndIndex < text.length() - 1) {
            argument = text.substring(cmdEndIndex+1);
        }

        handleCommand(update, chatId, argument);
        return true;
    }
}

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/SingleArgumentCommand.kt
package com.github.bumblebee.command

import com.github.bumblebee.bot.consumer.UpdateHandler
import com.github.telegram.domain.Update

abstract class SingleArgumentCommand : UpdateHandler {

    abstract fun handleCommand(update: Update, chatId: Long, argument: String?)

    override fun onUpdate(update: Update): Boolean {
        val chatId = update.senderId
        val text = update.message?.text ?: ""
        val cmdEndIndex = text.indexOf(' ')

        val argument = if (cmdEndIndex > 0 && cmdEndIndex < text.length - 1) {
            text.substring(cmdEndIndex + 1)
        } else null

        handleCommand(update, chatId, argument)
        return true
    }
}